MapboxNavigation/MD5.swift

```diff
@@ -0,0 +1,273 @@
+/*
+ * A JavaScript implementation of the RSA Data Security, Inc. MD5 Message
+ * Digest Algorithm, as defined in RFC 1321.
+ * Version 2.2 Copyright (C) Paul Johnston 1999 - 2009
+ * Other contributors: Greg Holt, Andrew Kepert, Ydnar, Lostinet
+ * Distributed under the BSD License
+ * See http://pajhome.org.uk/crypt/md5 for more info.
+ */
+
+/**
+ *  SwiftHash
+ *  Copyright (c) Khoa Pham 2017
+ *  Licensed under the MIT license. See LICENSE file.
+ */
+
+import Foundation
+
+// MARK: - Functions
+func MD5(_ input: String) -> String {
+    return hex_md5(input)
+}
+
+func hex_md5(_ input: String) -> String {
+    return rstr2hex(rstr_md5(str2rstr_utf8(input)))
+}
+
+func str2rstr_utf8(_ input: String) -> [CUnsignedChar] {
+    return Array(input.utf8)
+}
+
+func rstr2tr(_ input: [CUnsignedChar]) -> String {
+    var output: String = ""
+    
+    input.forEach {
+        output.append(String(UnicodeScalar($0)))
+    }
+    
+    return output
+}
+
+/*
+ * Convert a raw string to a hex string
+ */
+func rstr2hex(_ input: [CUnsignedChar]) -> String {
+    let hexTab: [Character] = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "A", "B", "C", "D", "E", "F"]
+    var output: [Character] = []
+    
+    for i in 0..<input.count {
+        let x = input[i]
+        let value1 = hexTab[Int((x >> 4) & 0x0F)]
+        let value2 = hexTab[Int(Int32(x) & 0x0F)]
+        
+        output.append(value1)
+        output.append(value2)
+    }
+    
+    return String(output)
+}
+
+/*
+ * Convert a raw string to an array of little-endian words
+ * Characters >255 have their high-byte silently ignored.
```

Comment:
We do care about the high byte. Add test cases for non-ASCII characters, all the way up to emoji like 🛡 (U+1F6E1). Although emoji would be somewhat contrived for an OSM-based data set, there are plenty of multibyte characters in OSM, especially in Chinese and Japanese.

Reply:
Looks like we’re no longer using this code, but a test case would still be good for peace of mind.